extensions/commands/deploy-lipo/_lipo.py

```diff
@@ -0,0 +1,95 @@
+import os
+import shutil
+from subprocess import run
+
+
+__all__ = ['is_macho_binary', 'lipo']
+
+# These are for optimization only, to avoid unnecessarily reading files.
+_binary_exts = ['.a', '.dylib']
+_regular_exts = [
+    '.h', '.hpp', '.hxx', '.c', '.cc', '.cxx', '.cpp', '.m', '.mm', '.txt', '.md', '.html', '.jpg', '.png'
+]
+
+
+def is_macho_binary(filename):
+    ext = os.path.splitext(filename)[1]
+    if ext in _binary_exts:
+        return True
+    if ext in _regular_exts:
+        return False
+    with open(filename, "rb") as f:
+        header = f.read(4)
+        if header == b'\xcf\xfa\xed\xfe':
+            # cffaedfe is Mach-O binary
+            return True
+        elif header == b'\xca\xfe\xba\xbe':
+            # cafebabe is Mach-O fat binary
+            return True
+        elif header == b'!<arch>\n':
+            # ar archive
+            return True
+    return False
+
+
+def copy_arch_file(src, dst, top=None, arch_folders=()):
+    if os.path.isfile(src):
+        if top and arch_folders and is_macho_binary(src):
+            # Try to lipo all available archs on the first path.
+            src_components = src.split(os.path.sep)
+            top_components = top.split(os.path.sep)
+            if src_components[:len(top_components)] == top_components:
+                paths = [os.path.join(a, *(src_components[len(top_components):])) for a in arch_folders]
+                paths = [p for p in paths if os.path.isfile(p)]
+                if len(paths) > 1:
+                    run(['lipo', '-output', dst, '-create'] + paths, check=True)
```

Comment:
Of course. This code started life as a wrapper outside of Conan. If this turns out to be useful, it might make sense to contribute it as `conan.tools.apple.lipo` in Conan itself.

Reply:
Actually, how do I get a conanfile here?  For a custom command it looks like I have to load it (and it might be .txt or .py).  I'm leaning towards the deployer, but that only has a dependency graph.  Can I get the conanfile from that?